lightning/mydump/loader.go

```diff
@@ -244,61 +263,38 @@ func (s *mdLoaderSetup) listFiles(dir string) error {
 			return nil
 		}
 
-		fname := strings.TrimSpace(f.Name())
-		lowerFName := strings.ToLower(fname)
-
-		info := fileInfo{path: path, size: f.Size()}
 		logger := log.With(zap.String("path", path))
 
-		var (
-			ftype         fileType
-			qualifiedName string
-		)
-		switch {
-		case strings.HasSuffix(lowerFName, "-schema-create.sql"):
-			ftype = fileTypeDatabaseSchema
-			qualifiedName = fname[:len(fname)-18] + "."
-
-		case strings.HasSuffix(lowerFName, "-schema.sql"):
-			ftype = fileTypeTableSchema
-			qualifiedName = fname[:len(fname)-11]
-
-			// ignore functionality :
-			// 		- view
-			//		- triggers
-		case strings.HasSuffix(lowerFName, "-schema-view.sql"),
-			strings.HasSuffix(lowerFName, "-schema-trigger.sql"),
-			strings.HasSuffix(lowerFName, "-schema-post.sql"):
-			logger.Warn("[loader] ignore unsupport view/trigger")
-			return nil
-		case strings.HasSuffix(lowerFName, ".sql"), strings.HasSuffix(lowerFName, ".csv"):
-			ftype = fileTypeTableData
-			qualifiedName = fname[:len(fname)-4]
-		default:
+		res := fr.Route(path)
+		if res == nil {
+			logger.Info("[loader] file is filtered by file router")
 			return nil
 		}
 
-		matchRes := tableNameRegexp.FindStringSubmatch(qualifiedName)
-		if len(matchRes) != 3 {
-			logger.Debug("[loader] ignore almost " + ftype.String() + " file")
-			return nil
+		info := fileInfo{
+			tableName: filter.Table{Schema: res.Schema, Name: res.Name},
+			path:      path,
+			size:      f.Size(),
 		}
-		info.tableName.Schema = matchRes[1]
-		info.tableName.Name = matchRes[2]
 
 		if s.loader.shouldSkip(&info.tableName) {
 			logger.Debug("[filter] ignoring table file")
+
 			return nil
 		}
 
-		switch ftype {
-		case fileTypeDatabaseSchema:
+		switch res.Type {
+		case SourceTypeSchemaSchema:
 			s.dbSchemas = append(s.dbSchemas, info)
-		case fileTypeTableSchema:
+		case SourceTypeTableSchema:
 			s.tableSchemas = append(s.tableSchemas, info)
-		case fileTypeTableData:
+		case SourceTypeSQL, SourceTypeCSV:
 			s.tableDatas = append(s.tableDatas, info)
 		}
+
```

Comment:
please add an "unsupported" warning (and ignore the file) if `Compression` is not none